modules/cli/src/main/scala/coursier/cli/install/SharedInstallParams.scala

```diff
@@ -32,10 +32,22 @@ final case class SharedInstallParams(
 
 object SharedInstallParams {
 
-  lazy val defaultDir = {
-    coursier.paths.CoursierPaths.dataLocalDirectory().toPath.resolve("bin")
+  private lazy val defaultDir0: Path = {
+
+    val fromEnv = Option(System.getenv("COURSIER_BIN_DIR")).filter(_.nonEmpty)
+      .orElse(Option(System.getenv("COURSIER_INSTALL_DIR")).filter(_.nonEmpty))
+
+    def fromProps = Option(System.getProperty("coursier.install.dir"))
+
+    def default = coursier.paths.CoursierPaths.dataLocalDirectory().toPath.resolve("bin")
+
+    fromEnv.orElse(fromProps).map(Paths.get(_))
+      .getOrElse(default)
   }
 
+  def defaultDir: Path =
+    defaultDir0
```

Comment:
Let's call `coursier.install.InstallDir.defaultDir`, instead of putting the logic computing the default directory here.

Reply:
Thanks, good point.